documentation/ProjectReference-Protocol.md

```diff
@@ -0,0 +1,69 @@
+
+`ResolveProjectReferences` does the bulk of the work, building the referenced projects and collecting their outputs.
+
+After the compiler is invoked, `GetCopyToOutputDirectoryItems` pulls child-project outputs into the current project to be copied to its output directory.
+
+When `Clean`ing the output of a project, `CleanReferencedProjects` ensures that referenced projects also clean.
+
+## Targets required to be referenceable
+
+These targets should exist in a project to be compatible with the common targets' `ProjectReference`. Some are called only conditionally.
+
+These targets are all defined in `Microsoft.Common.targets` and are defined in Microsoft SDKs. You should only have to implement them yourself if you require custom behavior or are authoring a project that doesn't import the common targets.
+
+<!--* `BuildGenerateSources` is run
+  * **Conditions**: only if `'$(BuildPassReferences)' == 'true'`.
+  * Rare in managed projects but common in C++ scenarios (for example, using IDL to generate header code).-->
+* `GetTargetFrameworkProperties` determines what properties should be passed to the main get-build-output target.
+  * **New** for MSBuild 15/Visual Studio 2017. Supports the cross-targeting feature allowing a project to have multiple `TargetFrameworks`.
+  * **Conditions**: only when metadata `SkipGetTargetFrameworkProperties` for each reference is not true.
+  * Skipped for `*.vcxproj` by default.
+* `GetTargetPath` should the path of the project's output, but _not_ build that output.
+  * **Conditions**: this is used for builds inside Visual Studio, but not on the command line.
+  * It's also used when the property `BuildProjectReferences` is `false`, manually indicating that all `ProjectReferences` are up to date and shouldn't be (re)built.
+* Default/explicitly specified targets do the full build and return an assembly to be referenced.
+  * If the `ProjectReference` defines the `Targets` metadata, it is used. If not, no target is passed, and the default target of the reference (usually `Build`) is built.
+* `GetNativeManifest` should return a manifest suitable for passing to the `ResolveNativeReferences` target.
+* `GetCopyToOutputDirectoryItems` should return the outputs of a project that should be copied to the output of a referencing project.
+* `Clean` should delete all outputs of the project.
+  * It is not called during a normal build, only during "Clean" and "Rebuild".
+
+## Other protocol requirements
+
+As with all MSBuild logic, targets can be added to do other work with `ProjectReference`s.
+
+In particular, NuGet depends on being able to identify referenced projects' package dependencies, and calls some targets that are imported through `Microsoft.Common.targets` to do so. At the time of writing this this is in [`NuGet.targets`](https://github.com/NuGet/NuGet.Client/blob/79264a74262354c1a8f899c2c9ddcaff58afaf62/src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets).
```

Comment:
Sorry, I said Xaml, but I should have said Appx, Microsoft.AppxPackage.targets is the specific thing that adds this requirement.